ci.sh

```diff
@@ -43,7 +43,7 @@ function run_tests {
     # optimizations up all the way, too).
     # Optimizations change diagnostics (mostly backtraces), so we don't check
     # them. Also error locations change so we don't run the failing tests.
-    MIRIFLAGS="${MIRIFLAGS:-} -O -Zmir-opt-level=4" MIRI_SKIP_UI_CHECKS=1 ./miri test -- tests/{pass,panic}
+    MIRIFLAGS="${MIRIFLAGS:-} -O -Zmir-opt-level=4 -Cdebug-assertions=yes" MIRI_SKIP_UI_CHECKS=1 ./miri test -- tests/{pass,panic}
```

Comment:
Why add the flag back here? I guess some tests rely on it? Please add a comment.

Reply:
Yeah. I considered doing something more clever, such `cfg(not(debug_assertions))` on the test cases, but the tests pretty explicitly call `debug_assert!` so it seemed very circular.